nni/experiment/experiment.py

```diff
@@ -193,8 +207,369 @@ def run(self, port: int = 8080, debug: bool = False) -> bool:
+        if key == 'max_trial_number':
+            return '?update_type=MAX_TRIAL_NUM'
+
+
+    def _update_experiment_profile(self, key: str, value: Any):
+        """
+        Update an experiment's profile
+
+        Parameters
+        ----------
+        key: str
+            One of `['trial_concurrency', 'max_experiment_duration', 'search_space', 'max_trial_number']`.
+        value: Any
+            New value of the key.
+        """
+        api = '/experiment{}'.format(self._get_query_type(key))
+        experiment_profile = self.get_experiment_profile()
+        experiment_profile['params'][key] = value
+        self._experiment_rest_put(self.port, api, experiment_profile)
+
+
+    def update_trial_concurrency(self, value: int):
+        """
+        Update an experiment's trial_concurrency
+
+        Parameters
+        ----------
+        value: int
+            New trial_concurrency value.
+        """
+        self._update_experiment_profile('trial_concurrency', value)
+
+
+    def update_max_experiment_duration(self, value: str):
+        """
+        Update an experiment's max_experiment_duration
+
+        Parameters
+        ----------
+        value: str
+            Strings like '1m' for one minute or '2h' for two hours.
+            SUFFIX may be 's' for seconds, 'm' for minutes, 'h' for hours or 'd' for days.
+        """
+        self._update_experiment_profile('max_experiment_duration', value)
+
+
+    def update_search_space(self, value: dict):
+        """
+        Update the experiment's search_space.
+        TODO: support searchspace file.
+
+        Parameters
+        ----------
+        value: dict
+            New search_space.
+        """
+        self._update_experiment_profile('search_space', value)
+
+
+    def update_max_trial_number(self, value):
+        """
+        Update an experiment's max_trial_number
+
+        Parameters
+        ----------
+        value: int
+            New max_trial_number value.
+        """
+        self._update_experiment_profile('max_trial_number', value)
+
+
+class TrialResult:
+    """
+    TrialResult stores the result information of a trial job.
+
+    Parameters
+    ----------
+    json_obj: dict
+        Json object that stores the result information.
+
+    Attributes
+    ----------
+    parameter: dict
+        Hyper parameters for this trial.
+    value: serializable object, usually a number, or a dict with key "default" and other extra keys
+        Final result.
+    trialJobId: str
+        Trial job id.
+    """
+    def __init__(self, json_obj):
+        self.parameter = None
+        self.value = None
+        self.trialJobId = None
+        for key in json_obj.keys():
+            setattr(self, key, json_obj[key])
+        self.value = json.loads(self.value)
+
+    def __repr__(self):
+        return "TrialResult(parameter: {} value: {} trialJobId: {})".format(self.parameter, self.value, self.trialJobId)
+
+
+class TrialMetricData:
```

Comment:
Maybe we should rewrite these classes with @dataclass?

Reply:
good suggestion, I will rewrite them